test/jdk/java/util/regex/NegativeArraySize.java

```diff
@@ -26,17 +26,17 @@
  * @bug 8223174
  * @summary Pattern.compile() can throw confusing NegativeArraySizeException
  * @requires os.maxMemory >= 5g
- * @run main/othervm NegativeArraySize -Xms5G -Xmx5G
+ * @run testng/othervm -Xms5G -Xmx5G NegativeArraySize
  */
 
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertThrows;
+
 import java.util.regex.Pattern;
 
 public class NegativeArraySize {
-    public static void main(String[] args) {
-        try {
-            Pattern.compile("\\Q" + "a".repeat(42 + Integer.MAX_VALUE / 3));
-            throw new AssertionError("expected to throw");
-        } catch (OutOfMemoryError expected) {
-        }
+    @Test
+    public static void testNegativeArraySize() {
+        assertThrows(OutOfMemoryError.class, () -> Pattern.compile("\\Q" + "a".repeat(42 + Integer.MAX_VALUE / 3)));
```

Comment:
One observation on this regex. Although the regex looks invalid because `\\Q` misses the pairing `\\E`, it can still be compiled (with a reasonable number of a's, of course). Moreover, the resulting pattern matches strings in a surprising way:

```
jshell> Pattern.compile("\\Qaaa").matcher("aaa").matches()
$1 ==> true
```

Reply:
Maybe that behavior is expected after all. From "Mastering Regular Expressions" by Jeffrey E.F. Friedl, 3rd Edition, p. 136:
> Literal-text span: `\Q...\E`
>
> First introduced with Perl, the special sequence `\Q...\E` turns off all regex meta-characters between them, except for `\E` itself. (If the `\E` is omitted, they are turned off until the end of the regex.)